openquake/hazardlib/gmice/base.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2012-2016 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module :mod:`openquake.hazardlib.gmice.base` defines base classes for
+different kinds of :class:`ground motion intensity conversion equations
+<GMICE>`.
+"""
+import abc
+import numpy as np
+from openquake.hazardlib.imt import MMI
+from openquake.hazardlib.gsim.base import GMPE, IPE
+
+class GMICE(GMPE):
+    """
+    Base class for a type of model known as a Ground Motion Intensity
+    Conversion Equation (GMIC). These models represent imt-specific
+    equations to convert between macroseismic intensity and known
+    ground motion parameters.
+
+    The majority of GMICEs are derived from regression models that are applied
+    in both directions:
+    GM -> Macroseismic Intensity
+    Macroseismic Intensity -> GM
+
+    The base class contains several key methods to apply the conversions.
+    """
+    # Defines the range of intensity values for which the GMICE applies
+    INTENSITY_RANGE = (0.0, 12.0) # Defaulting to full range
+
+    # Shakemap requires a "pretty" source name (e.g. Worden et al., (2012))
+    SOURCE_NAME = ""
+
+    # Shakemap requires a scale
+    SCALE = ""
+    
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        Doesn't function in this case but is an abstract method of the GMPE
+        class - so this is over-ridden to raise a not implemented error
+        """
+        raise NotImplementedError("Method 'get_mean_and_stddevs' not "
+                                  "available in GMICE object")
+
```

Comment:
I do not understand the docstring " Doesn't function in this case but is an abstract method of the GMPE". Why it does not function? You mean that this cannot be an `abc.abstractmethod` for some reason? Which reason?